### PR TITLE
Improve type hinting and favor statically typed parameters for Modality and Platform

### PR DESCRIPTION
**Pull request by bruno-f-cruz**
This PR refactors the `ManifestConfig` class by improving the type hinting for some of the fields.

This allows easier validation (without the need for custom validators) as well as auto-completion.

### CODE REVIEW

File: src/aind_watchdog_service/models/manifest_config.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class ManifestConfig(BaseModel):
         description="where to send data to on VAST",
         title="VAST destination and maybe S3?",
     )
-    modalities: Dict[str, List[str]] = Field(
+    modalities: Dict[Modality, List[str]] = Field(
```
Comment (arielleleon) [inline src/aind_watchdog_service/models/manifest_config.py:47]: Change Modality to Modality.ONE_OF. This should allow users to input a string that the Modality model can validate

File: src/aind_watchdog_service/models/manifest_config.py [hunk 2]
```diff
@@ -30,11 +30,11 @@ class ManifestConfig(BaseModel):
         description="Transfer time to schedule copy and upload. If None defaults to trigger the transfer immediately",  # noqa
         title="APScheduler transfer time",
     )
-    platform: str = Field(description="Platform type", title="Platform type")
+    platform: Platform = Field(description="Platform type", title="Platform type")
```
Comment (arielleleon) [inline src/aind_watchdog_service/models/manifest_config.py:33]: Change Platform to Platform.ONE_OF. This should allow users to input a string that the Platform model can validate

**Review by arielleleon — COMMENTED** [review]
@bruno-f-cruz Take a look at my comments. From what I understand you cannot type a platform (or modality) attribute as a Platform model type.

**Comment by bruno-f-cruz** [comment]
Relevant to this problem:

https://github.com/AllenNeuralDynamics/aind-data-schema/issues/960

I think we should code our own wrapper around the aind-data-schema for now that uses the same syntax

```python
class Modality(Enum, str):
    """Modality classes"""

    BEHAVIOR = modaloties.Modality.Behavior()
    BEHAVIOR_VIDEOS = modaloties.Modality.BehaviorVideos()

```

This solves a few problems:
1. Type hinting for users
2. easier runtime validation
3. Will likely be backwards compatible for users if aind-data-schema fixes the bug
 
It has a few cons:
Still requires a cast from our local type to the aind-data-schema-models type
If the problem in aind-data-schemas is not fixed, we will need to maintain this enumerator and keep adding entries to it.

**Comment by arielleleon** [comment]
I will add this wrapper to another issue. Does that work?

**Pull request merged**